reference-implementation/to-upstream-wpts/writable-streams/aborting.js

```diff
@@ -43,5 +43,193 @@ promise_test(t => {
+      writer.abort();
+      writer.write(4);
+      writer.write(5);
+    }).then(() => {
+      assert_equals(ws.events.length, 4, 'the stream should have received 4 events');
+      assert_equals(ws.events[1], 1, 'chunk should be 1');
+      assert_equals(ws.events[2], 'abort', 'abort should be called after write');
+      assert_equals(ws.events[3], undefined, 'no further writes should have happened');
+    });
+}, 'Aborting a WritableStream prevents further writes after any that are in progress');
+
+promise_test(() => {
+  const ws = new WritableStream({
+    abort() {
+      return 'Hello';
+    }
+  });
+
+  const writer = ws.getWriter();
+
+  const abortPromise = writer.abort('a');
+  return abortPromise.then(value => {
+    assert_equals(value, undefined, 'fulfillment value must be undefined');
+  });
+}, 'Fulfillment value of ws.abort() call must be undefined even if the underlying sink returns a non-undefined value');
+
+promise_test(t => {
+  const errorInSinkAbort = new Error('Sorry, it just wasn\'t meant to be.');
+  const ws = new WritableStream({
+    abort() {
+      throw errorInSinkAbort;
+    }
+  });
+
+  const writer = ws.getWriter();
+
+  return promise_rejects(t, errorInSinkAbort, writer.abort(undefined),
+    'rejection reason of abortPromise must be errorInSinkAbort');
+}, 'WritableStream if sink\'s abort throws, the promise returned by writer.abort() rejects');
+
+promise_test(t => {
+  const errorInSinkAbort = new Error('Sorry, it just wasn\'t meant to be.');
+  const ws = new WritableStream({
+    abort() {
+      throw errorInSinkAbort;
+    }
+  });
+
+  return promise_rejects(t, errorInSinkAbort, ws.abort(undefined),
+    'rejection reason of abortPromise must be errorInSinkAbort');
+}, 'WritableStream if sink\'s abort throws, the promise returned by ws.abort() rejects');
+
+test(() => {
+  let recordedReason;
+  const ws = new WritableStream({
+    abort(reason) {
+      recordedReason = reason;
+    }
+  });
+
+  const writer = ws.getWriter();
+
+  const passedReason = new Error('Sorry, it just wasn\'t meant to be.');
+  writer.abort(passedReason);
+
+  assert_equals(recordedReason, passedReason);
+}, 'Aborting a WritableStream passes through the given reason');
+
+promise_test(t => {
+  const ws = new WritableStream();
+  const writer = ws.getWriter();
+
+  const passedReason = new Error('Sorry, it just wasn\'t meant to be.');
+  writer.abort(passedReason);
+
+  return Promise.all([
+    promise_rejects(t, new TypeError(), writer.write(), 'writing should reject with the given reason'),
+    promise_rejects(t, new TypeError(), writer.close(), 'closing should reject with the given reason'),
+    promise_rejects(t, new TypeError(), writer.abort(), 'aborting should reject with the given reason'),
+    promise_rejects(t, new TypeError(), writer.closed, 'closed should reject with the given reason'),
+  ]);
+}, 'Aborting a WritableStream puts it in an errored state, with stored error equal to the abort reason');
```

Comment:
This test description is outdated; change it to "with a TypeError as the stored error"